openmc_plotter/plotmodel.py

```diff
@@ -650,9 +700,41 @@ def _do_op(array, tally_value, ax=0):
 
         return image_data, extents, data_min, data_max
 
+class ViewParam(openmc.lib.plot._PlotBase):
+    def __init__(self, origin=(0, 0, 0), width=10, height=10):
+        super().__init__()
+        # View Parameters
+        self.level = -1
+        self.origin = origin
+        self.width = width
+        self.height = height
+        self.h_res = 1000
+        self.v_res = 1000
+        self.basis = 'xy'
+        self.color_overlaps = False
+
+    def __eq__(self, other):
+        if (isinstance(other, ViewParam)):
+            for name in ['level', 'origin', 'width', 'height', 'h_res',
+                         'v_res', 'basis', 'color_overlaps']:
+                if self.__getattr__(name) != other.__getattr__(name):
+                    return False
+            return True
+        else:
+            return False
+
+    def __getattr__(self, name):
+        if name not in self.__dict__:
+            raise AttributeError('{} not in ViewParam dict'.format(name))
+        return self.__dict__[name]
+
+    def __setattr__(self, name, value):
+        super().__setattr__(name, value)
+        self.__dict__[name] = value
```

Comment:
I feel like this is not the correct way to do this, but if we want to check if two `ViewParam` objects are equal, we have to iterate over each attribute (see above `__eq__`) and that is not possible without also adding it to the `__dict__` explicitly.